src/org/etools/j1939_84/bus/RP1210Bus.java

```diff
@@ -97,7 +104,12 @@ public RP1210Bus(Adapter adapter, int address, boolean appPacketize) throws BusE
              adapter,
              address,
              appPacketize,
-             J1939_84.getLogger());
+             createBusAsyncLogger(J1939_84.getLogger()));
+    }
+
+    private static BusLogger createBusAsyncLogger(Logger logger) {
+        Executor exe = Executors.newSingleThreadExecutor();
+        return (severity, string, e) -> exe.execute(() -> logger.log(severity, string, e));
```

Comment:
It appears all places where this is used `e` is null

Reply:
Line 339:             getLogger().log(Level.SEVERE, "Failed to read RP1210", e);

Reply:
Ah.  I didn't think to look at lines that weren't changed.  

In that case, this looks good.